packages/flutter/test/widgets/form_test.dart

```diff
@@ -137,6 +137,90 @@ void main() {
     await checkErrorText('');
   });
 
+  testWidgets('IsValid returns true when all of the fields are valid', (WidgetTester tester) async {
+    final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+    const String validString = 'Valid string';
+    String validator(String s) => s == validString ? null : 'Error text';
+
+    Widget builder() {
+      return MaterialApp(
+        home: MediaQuery(
+          data: const MediaQueryData(devicePixelRatio: 1.0),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: Material(
+                child: Form(
+                  key: formKey,
+                  child: ListView(
+                    children: <Widget>[
+                      TextFormField(
+                        initialValue: validString,
+                        validator: validator,
+                        autovalidate: true
+                      ),
+                      TextFormField(
+                        initialValue: validString,
+                        validator: validator,
+                        autovalidate: true
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+
+    expect(formKey.currentState.isValid(), isTrue);
+  });
+
+  testWidgets('IsValid returns false when one of the fields is invalid', (WidgetTester tester) async {
+    final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+    const String validString = 'Valid string';
+    String validator(String s) => s == validString ? null : 'Error text';
+
+    Widget builder() {
+      return MaterialApp(
+        home: MediaQuery(
+          data: const MediaQueryData(devicePixelRatio: 1.0),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: Center(
+              child: Material(
+                child: Form(
+                  key: formKey,
+                  child: ListView(
+                    children: <Widget>[
+                      TextFormField(
+                        initialValue: '',
+                        validator: validator,
+                        autovalidate: true
+                      ),
+                      TextFormField(
+                        initialValue: validString,
+                        validator: validator,
+                        autovalidate: true
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(builder());
+
+    expect(formKey.currentState.isValid(), isFalse);
```

Comment:
This test should also assert that the state of any of the form fields didn't change and the widgets didn't rebuild with the error text.

Reply:
Done.